### ISSUE TITLE
[BUG] Heart Item Pickup Does Not Increase Life

### ISSUE THREAD

**Issue by Dream-no24** [dev-jjjjjeong-bin/Invaders-SDP#55]
## Problem Description

When a heart item is picked up, a log message "get heart item" appears, but the player's life does not increase as expected. This is causing an issue with the item functionality.

## Todo List

- [x] Fix the heart item pickup so that it properly increases the player's life

## Feature Description

Ensure that when a heart item is picked up, it correctly increases the player's life, and the appropriate log message is displayed.

## Additional Information

The heart item pickup currently triggers a log message, but there seems to be a disconnect between the item pickup and the life increment functionality.

**Comment by ise-e** [comment]
After checking, it was confirmed that physical strength was recovered properly.

Is it true that you acquired the item while the player's health was low?

**Comment by Dream-no24** [comment]
Oh, I opened the issue after confirming that even if I ate the item when my physical strength was 2, my physical strength would not increase. But I confirmed that the heart item was working normally.

I think the file of src/inventory_develop/ItemBarrierAndHeart.java that is in charge of this is found to have been added in #51 PR, and it seems that such a bug appeared unreflected in our forked repo. 

Sorry for the confusion, thank you for checking!
